### ISSUE TITLE
Adding numbers to (sub)sections?

### ISSUE THREAD

**Issue by TooHuman** [cambridge/thesis#34]
Hi, I was just wondering how you would edit this template to include numbering in the section and subsection headings. I realise these are already present in the ToC but is there an easy way to enable them?

Using the simple or clean template that was provided by the way.

Thanks,
Mike.

**Comment by mgkuhn** [comment]
\section{heading} as explained in Leslie Lamport's LaTeX book?

**Comment by urbas** [comment]
Hey @TooHuman, I don't think you have to enable them. They are enabled by default.

Perhaps you had something else in mind?

**Comment by TooHuman** [comment]
I'm using the template provided in Samples/clean/ and when doing so, the chapters are numbered, but sections and subsections are not numbered on the actual pages they appear. From the pdf supplied with the normal version, it appears that there is no numbering for the section headings either? As in it displays "Chapter 1", but not "1.1 Introduction", just "Introduction". I hope this is clearer.

**Comment by mgkuhn** [comment]
[This example document](https://dl.bintray.com/matej/cam-thesis/sample-clean.pdf) ([source file](https://github.com/cambridge/thesis/blob/master/Samples/clean/11Introduction.tex)) has no section in chapter 1. It has only `\chapter{Introduction}` “Chapter 1: Introduction”, and that number is displayed correctly (one line above).

If you do want to add a section, use `\section{...}`. If you don't want to add a chapter, don't use `\chapter{...}`. You don't have to use `\chapter{...}` at all if you don't like chapters; using only sections is also perfectly fine. Sections are automatically numbered, but if you really wanted an unnumbered section then use `\section*{...}`.

This is all explained in Leslie Lamport: LaTeX – a document preparation system, section 2.2.3. It has nothing to do with this template. (Do read the LaTeX manual carefully. LaTeX is not software that can be used without reading the manual, by merely copying examples. Doing so results in well-known and obvious beginners mistakes that most examiners don't like.)

**Comment by TooHuman** [comment]
Thank you for the.. somewhat patronising response.

Here is a screenshot of the clean template in use, with added sections (and on the next page subsections) which are **not** numbered. I do understand how numbered and unnumbered sections work in latex, and I am using `\section{...}` here but as you can see, these are not numbered in the same way that the chapter is.
[Screenshot](https://www.dropbox.com/s/d3qb2ibruypyk4c/cambridge_thesis_sections.png?dl=0)

Thanks.

**Comment by urbas** [comment]
That's interesting. Many apologies for the misunderstanding, but it looks like you might be doing something differently to what the plain Sample Clean does.

Please take a look at the 3rd chapter in this pdf: https://dl.bintray.com/matej/cam-thesis/sample-clean.pdf

You'll see that the sections are actually numbered. 

Could you check whether you have added any configuration in your preamble that could change this behaviour?

**Comment by TooHuman** [comment]
Replacing the preamble with the version in the clean directory under Samples doesn't seem to have changed anything. Think I'll try and build a fresh project and see if it makes any difference.

**Comment by TooHuman** [comment]
So when downloading the thesis directory, extracting, and copying the clean template into the root directory, then building, I get the following [PDF](https://www.dropbox.com/s/6gjo2tsebse7474/thesis.pdf?dl=0). Again, no sections are numbered, so I must be doing something fundamentally wrong.

**Comment by mgkuhn** [comment]
I was able to reproduce your problem on Ubuntu 16.04 with

```
$ git clone https://github.com/cambridge/thesis.git
$ cd thesis/Samples/clean/
$ export TEXINPUTS=.:../..:
$ pdflatex thesis ; pdflatex thesis
$ okular thesis.pdf
```

Page 21 has a section "Section" without number.

**Comment by TooHuman** [comment]
I have been building on Linux Mint 18.1. Thanks for the reproduction.

**Comment by urbas** [comment]
Aha, so this is a bug. I'll try to fix it asap.

**Comment by mgkuhn** [comment]
The culprit are the lines following \RequirePackage{titlesec} in 01Preamble.tex. Proof: replace them with

```
\newcommand{\PreContentTitleFormat}{}
\newcommand{\ContentTitleFormat}{}
```

and the section numbers are back.

I suggest reading `texdoc titlesec` carefully to understand what all this is meant to do. Never use a LaTeX package where you haven't read the documentation. (Patronizing advice, may be. But it's based on nearly three decades of experience with observing people blindly copying the use of LaTeX packages they don't understand).

**Comment by urbas** [comment]
Thank you for the reproduction! (And the patience.) I did not notice this bug when building on Ubuntu 16.04.

**Comment by TooHuman** [comment]
This doesn't seem to solve the problem for me on a fresh build?

**Comment by TooHuman** [comment]
I can't reproduce this fix on a fresh build of the clean template.

**Comment by urbas** [comment]
Hey @TooHuman, could try the fix proposed in pull request #36?

**Comment by TooHuman** [comment]
This certainly seems to fix the problem. Sections and subsections are now numbered appropriately. Thanks for the help with this!
